url90b6689baa_adsonrocha_PD321_tgz-pJ8-MyFuzzyLiteClassJ8/src/com/fuzzylite/defuzzifier/Tsukamoto.java

```diff
@@ -0,0 +1,107 @@
+/*   Copyright 2013 Juan Rada-Vilela
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+package com.fuzzylite.defuzzifier;
+
+import com.fuzzylite.FuzzyLite;
+import com.fuzzylite.Op;
+import com.fuzzylite.term.Ramp;
+import com.fuzzylite.term.SShape;
+import com.fuzzylite.term.Sigmoid;
+import com.fuzzylite.term.Term;
+import com.fuzzylite.term.Thresholded;
+import com.fuzzylite.term.ZShape;
+
+/**
+ *
+ * @author jcrada
+ */
+public class Tsukamoto {
+
+    public static double tsukamoto(Thresholded term, double minimum, double maximum) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 13. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 88-104 into a separate method.